### ISSUE TITLE
24-06-19 Meeting w/ Camille & Elisa

### ISSUE THREAD

**Issue by cmaumet** [gkiar/analytical-stability#9]
Lots of things were discussed but I just wanted to keep a link to this map that @gkiar identified as outlier/to be excluded: https://neurovault.org/images/100001/

**Comment by gkiar** [comment]
#### Model loss function
- still use MSE as base for loss function
- consider adding L1 regularization for minimizing the number of non-zero voxels at the conv4 layer

AND/OR

- consider regularizing the bottleneck (conv4) layer with discrete KL divergence? w.r.t. what distribution?

For additional info on spare autoencoders, look at: https://www.jeremyjordan.me/autoencoders/

#### Data quality
- consider filtering maps by texture-based-thresholding (must not be too smooth)
- consider only reconstructing group maps at present, because we have more faith in them.

#### HCP Tasks
- Found fsf and event files for motor task
- [x] @cmaumet to enumerate a check-list of all things that Greg should populate for each task
- [ ] @gkiar to track down equivalent files for all tasks of interest and populate the checklist from Camille with multiple examples for each paradigm to identify how consistent they are across sessions/subjects/runs

**Comment by gkiar** [comment]
[Example](https://spandan-madan.github.io/A-Collection-of-important-tasks-in-pytorch/#SECTION-5---CUSTOM-LOSS-FUNCTIONS) for creating custom loss functions in PyTorch.

**Comment by cmaumet** [comment]
Check-list to populate for each task:
 - List of regressors of interests with their timing (onset, duration, possibly height)
 - Contrast(s) of interest as a linear combination of the regressors.

**Comment by gkiar** [comment]
https://biobank.ctsu.ox.ac.uk/crystal/docs/brain_mri.pdf

**Comment by gkiar** [comment]
@cmaumet I have a few questions looking for more detail on the checklist you provided.... Please forgive stupid questions and bear in mind I've *never* done an fMRI analysis, let alone tried to replicate one 😅 

- regressors of interest: to confirm, this is an "event"? I know from the files in the EV folder how to spot onset and duration, but where would height/intensity be encoded?
- what contrasts are of interest? We noted that they define a ton, and you seemed to not be interested in all of them. Is there a rule of thumb I can follow? Essentially do I just look for those which are not averages or negatives and record those? Or do we care about every single contrast? For example, in the model below do I just care about c1-c7? If so/not, why?

![image](https://user-images.githubusercontent.com/4883288/60169832-bdeb0580-9807-11e9-9b98-6435dac0fe8c.png)

**Comment by cmaumet** [comment]
@gkiar: happy to answer any question you may have :)

> regressors of interest: to confirm, this is an "event"? I know from the files in the EV folder how to spot onset and duration, but where would height/intensity be encoded?

Yes "regressors"="events". 

The 'height' will usually be constant and equal to 1. Here is FSL's documentation on how they encode events in their 3-column event files: https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FEAT/FAQ#A.22FULL.22_MODELLING (3rd column in 'height'). That's probably the files you should start off to retrieve onsets/durations/heights (events in the design matrix have been convolved with the HRF and are not the original ones any more). Is that what you are looking at?

> what contrasts are of interest?

As discussed, there is no automatic rule to retrieve a contrast of interest as they could take many different forms. My recommendation is to look at the figures of the HCP paper and decide for each of the 2 paradigms we will be looking at, which contrast(s) we want to keep.

Hope that makes sense! Let me know if you need more :)
